test/testutil/tests.c

```diff
@@ -39,37 +58,238 @@
  *
  * calling test6(3, "oops") will return 0 and produce out along the lines of:
  *      FAIL oops: (int) value 3 is not 6\n
- *
- * It general, test_fail_message should not be called directly.
  */
 static void test_fail_message(const char *prefix, const char *file, int line,
-                              const char *type, const char *fmt, ...)
-            PRINTF_FORMAT(5, 6);
-
-static void test_fail_message_va(const char *prefix, const char *file, int line,
-                                 const char *type, const char *fmt, va_list ap)
+                              const char *type, const char *, const char *,
+                              const char *, const char *fmt, ...)
```

Comment:
Please keep the names `left`, `right` and `op` here.  They do serve as documentation.